File: src/main/java/com/refactortalentscrm/service/nullService.java
package com.refactortalentscrm.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.refactortalentscrm.domain.${className};
import com.refactortalentscrm.dto.${className}DTO;
import com.refactortalentscrm.dto.${className}SearchDTO;
import com.refactortalentscrm.dto.${className}PageDTO;
import com.refactortalentscrm.dto.${className}ConvertCriteriaDTO;
import com.refactortalentscrm.service.GenericService;
import com.refactortalentscrm.dto.common.RequestDTO;
import com.refactortalentscrm.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ${className}Service extends GenericService<${className}, Integer> {

	List<${className}> findAll();

	ResultDTO add${className}(${className}DTO ${classNameCamel}DTO, RequestDTO requestDTO);

	ResultDTO update${className}(${className}DTO ${classNameCamel}DTO, RequestDTO requestDTO);

    Page<${className}> getAll${classNamePlural}(Pageable pageable);

    Page<${className}> getAll${classNamePlural}(Specification<${className}> spec, Pageable pageable);

	ResponseEntity<${className}PageDTO> get${classNamePlural}(${className}SearchDTO ${classNameCamel}SearchDTO);
	
	List<${className}DTO> convert${classNamePlural}To${className}DTOs(List<${className}> ${classNameCamelPlural}, ${className}ConvertCriteriaDTO convertCriteria);

	${className}DTO get${className}DTOById(Integer ${classNameCamel}Id);







}

File: refactortalentscrm_ui/src/services/nullService.js
import http from "../http-common";

class ${className}Service {
  getAll${classNamePlural}(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/${classNameCamel}/${classNameCamelPlural}`, searchDTO);
  }

  get(${classNameCamel}Id) {
    return this.getRequest(`/${classNameCamel}/${${classNameCamel}Id}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/${classNameCamel}?field=${matchData}`, null);
  }

  add${className}(data) {
    return http.post("/${classNameCamel}/add${className}", data);
  }

  update(data) {
  	return http.post("/${classNameCamel}/update${className}", data);
  }
  
  uploadImage(data,${classNameCamel}Id) {
  	return http.postForm("/${classNameCamel}/uploadImage/"+${classNameCamel}Id, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ${className}Service();
